internal/agent/agent.go

```diff
@@ -1208,11 +1209,21 @@ func (c *context) SendEvent(event sample.Event, entityKey entity.Key) {
 	}
 }
 
+// Decides wether an event will be included or not.
+// This kind of filtering only applies to ProcessSamples,
+// so that's what we check here.
 func (c *context) IncludeEvent(event any) bool {
-	shouldInclude := c.shouldIncludeEvent(event)
-	shouldExclude := c.shouldExcludeEvent(event)
-
-	return shouldInclude || !shouldExclude
+	switch event.(type) {
+	// rule is applied to process samples
+	case *process_sample_types.ProcessSample, *process_sample_types.FlatProcessSample:
+		shouldInclude := c.shouldIncludeEvent(event)
+		shouldExclude := c.shouldExcludeEvent(event)
+
+		return shouldInclude || !shouldExclude
+	default:
+		// other samples are included
+		return true
+	}
```

Comment:
nit: I actually liked this approach to evaluate the type of sample even before calling the matcher functions.

Reply:
I have retrieved it in https://github.com/newrelic/infrastructure-agent/pull/1943/commits/abdd6961729382184feb1424670eeb32af6fa701! LMKWYT